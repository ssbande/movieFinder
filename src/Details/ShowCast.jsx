import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Header, Card, Image } from 'semantic-ui-react';

import { createImageUrl, addDefaultSrc } from '../utils/helpers';

const ShowCast = ({ cast, handleCardClick }) => {
  return <div style={{ marginTop: 20 }}>
    <Header size='huge' color='yellow' inverted >Cast</Header>
    <Grid stackable columns={4}>
      {cast.map(character => {
        const {
          id,
          profile_path,
          name,
          character: characterName
        } = character;

        return (
          <Grid.Column key={id}>
            <Card onClick={() => handleCardClick(id)} href="">
              <Card.Content>
                <Image
                  onError={addDefaultSrc}
                  size='big'
                  style={{ marginBottom: 10 }}
                  src={createImageUrl(profile_path)}
                />
                <Card.Header>{name}</Card.Header>
                <Card.Meta>{characterName}</Card.Meta>
              </Card.Content>
            </Card>
          </Grid.Column>
        )
      })}
    </Grid>
  </div>
}

ShowCast.propTypes = {
  cast: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired, 
    profile_path: PropTypes.string, 
    name: PropTypes.string, 
    character: PropTypes.string
  })), 
  handleCardClick: PropTypes.func.isRequired
}

export default ShowCast;