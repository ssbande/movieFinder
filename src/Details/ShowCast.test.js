import React from 'react';
import { shallow } from 'enzyme';

import ShowCast from './ShowCast';

const setup = () => ({
  props: {
    cast: [{
      id: 1001,
      profile_path: 'profile_path',
      name: 'name1',
      character: 'character1'
    }],
    handleCardClick: jest.fn(),
  }
})

describe('Show Cast', () => {
  it('should render correctly', () => {
    const { props } = setup();
    const wrapper = shallow(<ShowCast {...props} />);
    expect(wrapper).toMatchSnapshot();
  })
})