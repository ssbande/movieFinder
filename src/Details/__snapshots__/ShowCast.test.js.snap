// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Cast should render correctly 1`] = `ShallowWrapper {}`;
